libpod/runtime_img.go

```diff
@@ -139,9 +140,21 @@ func (r *Runtime) NewImage(name string) Image {
 
 // GetImageID returns the image ID of the image
 func (k *Image) GetImageID() (string, error) {
+	// If the ID field is already populated, then
+	// return it.
 	if k.ID != "" {
 		return k.ID, nil
 	}
+	// If we have the name of the image locally, then
+	// get the image and returns its ID
+	if k.LocalName != "" {
+		img, err := k.runtime.GetImage(k.LocalName)
+		if err == nil {
+			return img.ID, nil
+		}
+	}
+	// If neither the ID is known and no local name
+	// is know, we search it out.
```

Comment:
could you instead drop 148-155 and just do 
```
image := k.LocalName
if image != "" {
	image, _ = k.GetFQName()
}
img, err := k.runtime.GetImage(image)
```

Reply:
k.LocalName is not always set for local creation.  And the possible costly function that includes network usage is GetFQName ...

Reply:
Ah, ok, then yeah keep it as is if GetFQName is the suspect, it's a bit better this way.